dev/tests/eval-tests.nix

```diff
@@ -2,59 +2,257 @@
+      tooEmptyResultTried1TestTried = builtins.tryEval (tooEmptyResultTried == evalTests.tooEmptyResultTried1);
+    in
+    assert tooEmptyResultTried0TestTried == evalTests.tooEmptyResultTried0TestTried;
+    assert tooEmptyResultTried1TestTried == evalTests.tooEmptyResultTried1TestTried;
+    ok;
+  tooEmptyTestsResult = evalTests.runTooEmptyTests "ok";
+
+  example1 = evalTests.callFlake {
+    inputs.flake-parts = evalTests.flake-parts;
+    inputs.self = { };
+    outputs = inputs: inputs.flake-parts.lib.mkFlake { inherit inputs; } {
       systems = [ "a" "b" ];
       perSystem = { system, ... }: {
-        packages.hello = pkg system "hello";
+        packages.hello = evalTests.pkg system "hello";
       };
     };
+  };
+  weakEvalTests.example1Result = {
+    apps = { a = { }; b = { }; };
+    checks = { a = { }; b = { }; };
+    devShells = { a = { }; b = { }; };
+    formatter = { };
+    legacyPackages = { a = { }; b = { }; };
+    nixosConfigurations = { };
+    nixosModules = { };
+    overlays = { };
+    packages = {
+      a = { hello = evalTests.pkg "a" "hello"; };
+      b = { hello = evalTests.pkg "b" "hello"; };
+    };
+  };
+  strongEvalTests.example1Result = let
+    _type = "flake";
+    inputs.flake-parts = evalTests.flake-parts;
+    inputs.self = { };
+    outputs = evalTests.weakEvalTests.example1Result;
+    sourceInfo.outPath = "/unknown_eval-tests_flake";
+    result = outputs // sourceInfo // { inherit _type inputs outputs sourceInfo; };
+  in result;
+  runExample1Tests = ok:
+    assert evalTests.example1 == evalTests.example1Result;
+    ok;
+  example1TestsResult = evalTests.runExample1Tests "ok";
+
+  # This test case is a fun one. In the REPL, try `exhibitInfiniteRecursion.*`.
+  # In the case that `mkFlake` *isn't* called from a flake, `inputs.self` is
+  # unlikely to refer to the result of the `mkFlake` evaluation. If
+  # `inputs.self` isn't actually self-referential, evaluating attribute values
+  # of `self` is not divergent. Evaluation of `self.outPath` is useful for
+  # paths in documentation & error messages. However, if that evaluation occurs
+  # in a `builtins.addErrorContext` message forced by an erroring `self`, both
+  # `self` will never evaluate *and* `builtins.toString self.outPath` must
+  # evaluate, causing Nix to instead throw an infinite recursion error. Even
+  # just `inputs.self ? outPath` throws an infinite recursion error.
+  # (`builtins.tryEval` can only catch errors created by `builtins.throw` or
+  # `builtins.assert`, so evaluation is guarded with
+  # `exhibitingInfiniteRecursion` here to keep `runTests` from diverging.)
+  # In this particular case, `mkFlake` evaluates `self ? outPath` to know if the
+  # default module location it provides should be generic or specific. As
+  # explained, this evaluation is unsafe under an uncatchably divergent `self`.
+  # Thus, `outPath` cannot be safely sourced from `self` at the top-level.
```

Comment:
:tada: I can confirm that this now works as it should, since #192 or #194.

<details><summary>The new, informative error message</summary>

```
«error: error:
       … while evaluating the attribute 'strongEvalTests.nonexistentOption'

         at /home/user/h/flake-parts/dev/tests/eval-tests.nix:209:3:

          208|   # exhibiting infinite recursion, the flake is made equivalent to `empty`.
          209|   strongEvalTests.nonexistentOption = evalTests.callFlake {
             |   ^
          210|     inputs.flake-parts = evalTests.flake-parts;

       … in the left operand of the update (//) operator

         at /home/user/h/flake-parts/dev/tests/eval-tests.nix:33:24:

           32|       outputs = flake.outputs (inputs // { self = result; });
           33|       result = outputs // sourceInfo // {
             |                        ^
           34|         inherit inputs outputs sourceInfo;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: The option `nonexistentOption' does not exist. Definition values:
       - In `/home/user/h/flake-parts/dev/tests/eval-tests.nix': null»
```

</details>